src/Components/Web.Extensions/src/ProtectedBrowserStorage/ProtectedBrowserStorage.cs

```diff
@@ -0,0 +1,202 @@
+            _dataProtectionProvider = dataProtectionProvider ?? throw new ArgumentNullException(nameof(dataProtectionProvider));
+        }
+
+        /// <summary>
+        /// <para>
+        /// Asynchronously stores the specified data.
+        /// </para>
+        /// <para>
+        /// Since no data protection purpose is specified with this overload, the purpose is derived from <paramref name="key"/> and the store name. This is a good default purpose to use if the keys come from a fixed set known at compile-time.
+        /// </para>
+        /// </summary>
+        /// <param name="key">A <see cref="string"/> value specifying the name of the storage slot to use.</param>
+        /// <param name="value">A JSON-serializable value to be stored.</param>
+        /// <returns>A <see cref="ValueTask"/> representing the completion of the operation.</returns>
+        public ValueTask SetAsync(string key, object value)
+            => SetAsync(CreatePurposeFromKey(key), key, value);
+
+        /// <summary>
+        /// Asynchronously stores the supplied data.
+        /// </summary>
+        /// <param name="purpose">A string that defines a scope for the data protection. The protected data can only be unprotected by code that specifies the same purpose.</param>
+        /// <param name="key">A <see cref="string"/> value specifying the name of the storage slot to use.</param>
+        /// <param name="value">A JSON-serializable value to be stored.</param>
+        /// <returns>A <see cref="ValueTask"/> representing the completion of the operation.</returns>
+        public ValueTask SetAsync(string purpose, string key, object value)
+        {
+            if (string.IsNullOrEmpty(purpose))
+            {
+                throw new ArgumentException("Cannot be null or empty", nameof(purpose));
+            }
+
+            if (string.IsNullOrEmpty(key))
+            {
+                throw new ArgumentException("Cannot be null or empty", nameof(key));
+            }
+
+            return SetProtectedJsonAsync(key, Protect(purpose, value));
+        }
+
+        /// <summary>
+        /// <para>
+        /// Asynchronously retrieves the specified data.
+        /// </para>
+        /// <para>
+        /// The first value in the <see cref="ValueTask"/> result indicates whether the data was retrieved successfully.
+        /// </para>
+        /// <para>
+        /// Since no data protection purpose is specified with this overload, the purpose is derived from <paramref name="key"/> and the store name. This is a good default purpose to use if the keys come from a fixed set known at compile-time.
+        /// </para>
+        /// </summary>
+        /// <param name="key">A <see cref="string"/> value specifying the name of the storage slot to use.</param>
+        /// <returns>A <see cref="ValueTask"/> representing the completion of the operation.</returns>
+        public ValueTask<(bool success, T result)> TryGetAsync<T>(string key)
```

Comment:
I don't think we return value tuples as part of the API surface. It also does not work in a typical `if (TryGetValue)` code block. It might be better to return a result type here. `ValueTask<ProtectedBrowserStorageResult> GetAsync`. We could possibly move `GetValueOrDefaultAsync` to it.

Reply:
I've gone ahead and changed `TryGetValue` to `GetAsync` that returns a `ValueTask<ProtectedBrowserStorageResult>`. I've also removed `GetValueOrDefaultAsync` since the user could easily implement their own helper that does this if they really need it. But, if you or others think it's worth leaving in, I could easily add it back.